subsys/bluetooth/audio/bap_stream.c

```diff
@@ -302,27 +300,38 @@ bool bt_audio_valid_qos_pref(const struct bt_audio_codec_qos_pref *qos_pref)
 
 	/* The absolute minimum and maximum values of pref_pd_min and pref_pd_max are implicitly
 	 * checked using the bounds of pd_min and pd_max, so we can just compare the preferences
-	 * to the min and max values that have been bound checked already
+	 * to the min and max values that have been bound checked already.
+	 * If pref_pd_min is not BT_AUDIO_PD_PREF_NONE then it shall be >= pd_min
 	 */
-	if (!IN_RANGE(qos_pref->pref_pd_min, qos_pref->pd_min, qos_pref->pd_max)) {
+	if (qos_pref->pref_pd_min != BT_AUDIO_PD_PREF_NONE &&
+	    !IN_RANGE(qos_pref->pref_pd_min, qos_pref->pd_min, qos_pref->pd_max)) {
 		LOG_DBG("Invalid combination of pref_pd_min %u, pd_min %u and pd_max: %u",
 			qos_pref->pref_pd_min, qos_pref->pd_min, qos_pref->pd_max);
 
 		return false;
 	}
 
-	if (qos_pref->pref_pd_max < qos_pref->pref_pd_min) {
+	if (qos_pref->pref_pd_max != BT_AUDIO_PD_PREF_NONE &&
+	    qos_pref->pref_pd_max < qos_pref->pref_pd_min) {
 		LOG_DBG("Invalid combination of pref_pd_min %u and pref_pd_max: %u",
 			qos_pref->pref_pd_min, qos_pref->pref_pd_max);
 
 		return false;
 	}
+	/* If pref_pd_max is not BT_AUDIO_PD_PREF_NONE then it shall pref_pd_max <= pd_max as per
+	 * the ASCS spec, but should also be >= MAX(pd_min, pref_pd_min)
+	 */
+	if (qos_pref->pref_pd_max != BT_AUDIO_PD_PREF_NONE) {
```

Comment:
Should just be:
	
	if (qos_pref->pref_pd_max != BT_AUDIO_PD_PREF_NONE) {
		if (!IN_RANGE(qos_pref->pref_pd_max, qos_pref->pd_min, qos_pref->pd_max)) {
		
With the change requested previously it has already been validated that 
qos_pref->pref_pd_max <= qos_pref->pref_pd_min